apollo-gradle-plugin/src/main/groovy/com/apollographql/apollo/gradle/ApolloSystemCodegenGenerationTask.groovy

```diff
@@ -32,4 +37,24 @@ class ApolloSystemCodegenGenerationTask extends AbstractExecTask<ApolloSystemCod
       super.exec()
     }
   }
+
+  private static verifySystemApolloCodegenVersion() {
+    println("Verifying system 'apollo-codegen' version (executing command 'apollo-codegen --version') ...")
+    try {
+      StringBuilder output = new StringBuilder()
+      Process checkGlobalApolloCodegen = "apollo-codegen --version".execute()
+      checkGlobalApolloCodegen.consumeProcessOutput(output, null)
+      checkGlobalApolloCodegen.waitForOrKill(5000)
+
+      def version = output.toString().trim()
+      if (version == APOLLOCODEGEN_VERSION) {
+        println("Found required 'apollo-codegen@$APOLLOCODEGEN_VERSION' version.")
+        println("Skip apollo-codegen installation.")
+      } else {
+        throw new GradleException("Required 'apollo-codegen@$APOLLOCODEGEN_VERSION' version but found: $version. Consider disabling `apollographql.useGlobalApolloCodegen` property.")
+      }
+    } catch (Exception exception) {
+      throw new GradleException("Failed to verify system 'apollo-codegen' version. Consider disabling `apollographql.useGlobalApolloCodegen` property.", exception)
+    }
+  }
```

Comment:
nit: should it be used `logger.info`  instead of `println` ?

Reply:
Addressed in https://github.com/apollographql/apollo-android/pull/1451/commits/b5eb9ba6d54ac6ebe211fbdaf5ba6f538f1cab80